.github/workflows/docsgen-cli.yml

```
@@ -0,0 +1,31 @@
+name: Generate Documentation
+
```

```suggestion

permissions:
  contents: write

```

I don't recall what our org settings are but this may be needed to be able to push changes from an action; doesn't hurt to have it either way